include/frozen/map.h

```diff
@@ -323,6 +418,16 @@ constexpr auto make_map(std::array<std::pair<T, U>, N> const &items) {
   return map<T, U, N>{items};
 }
 
+template <typename T, typename U, typename Compare, std::size_t N>
+constexpr auto make_map(std::pair<T, U> const (&items)[N]) {
+  return map<T, U, N, Compare>{items};
+}
+
+template <typename T, typename U, typename Compare, std::size_t N>
+constexpr auto make_map(std::array<std::pair<T, U>, N> const &items) {
+  return map<T, U, N, Compare>{items};
+}
+
```

Comment:
Why not making `Compare` the last template parameter, with a default value, which would remove the need for these two overloads?

Reply:
I thought the `make_map` methods were helpful in removing the need to specify `N` -- ie. have the compiler deduce `N`. No?
With:
```
template <typename T, typename U, std::size_t N, typename Compare = std::less<T>>
constexpr auto make_map(std::array<std::pair<T, U>, N> const &items) {
  return map<T, U, N, Compare>{items};
}
```
You'd have to specify `N` 
```
  constexpr auto frozen_map3 = frozen::make_map<int, int, 128, std::less<void>>(std::array<std::pair<int, int>, 128>{{INIT_SEQ}});
```
instead of 
```
  constexpr auto frozen_map3 = frozen::make_map<int, int, std::less<void>>(std::array<std::pair<int, int>, 128>{{INIT_SEQ}});
```


To be honest I'd much prefer not specifying `T` and `U` either, as in:
```
  constexpr auto frozen_map3 = frozen::make_map<std::less<void>>(std::array<std::pair<int, int>, 128>{{INIT_SEQ}});
```
What do you think about?
```
template <typename Compare, typename T, typename U, std::size_t N>
constexpr auto make_map(std::array<std::pair<T, U>, N> const &items) {
  return map<T, U, N, Compare>{items};
}
```
so that one can write:
```
  constexpr auto frozen_map3 = frozen::make_map<std::less<void>>(std::array<std::pair<int, int>, 128>{{INIT_SEQ}});
```

Reply:
> What do you think about?
>
> template <typename Compare, typename T, typename U, std::size_t N>
> constexpr auto make_map(std::array<std::pair<T, U>, N> const &items) {
> return map<T, U, N, Compare>{items};
>}
>
>so that one can write:
>
>  constexpr auto frozen_map3 = frozen::make_map<std::less<void>>(std::array<std::pair<int, int>, 

I like that idea. Let's merge this PR and rewrite the argument order that way in another one.